ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -452,7 +452,13 @@ public boolean beforeCompletion (Xid xid) throws javax.transaction.SystemExcepti
 
     public Transaction getTransaction(Xid xid) throws XAException {
         // first see if the xid is a root coordinator
-        return com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
+        Transaction transaction = com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple
```

Comment:
I think there is a good chance there is an existing bug here. I think it is wrong to try to get an ArjunaCore TransactionImple here, it should be a JTS one and could be a copy/paste error here: https://github.com/jbosstm/narayana/commit/db5e42340d636c89d348ae2bfec761809b8e0b70#diff-ea4bafab11ba2836771979e805769d8bR445 what do you think @mmusgrov